src/wp-includes/class-wp-date-query.php

```diff
@@ -312,9 +312,9 @@ public function validate_date_values( $date_query = array() ) {
 			 * If multiple years are provided (as in a BETWEEN), use the first one.
 			 */
 			if ( is_array( $date_query['year'] ) ) {
-				$_year = reset( $date_query['year'] );
+				$_year = intval( reset( $date_query['year'] )) ;
```

Comment:
`reset()` can return `false`. I think the condition wrapping this should probably be adjusted to also check for a non-empty array to avoid this.

Also, please do not use `intval()` except as a callback. A simple `(int)` cast is better.